### PR TITLE
docs: remove needless scripts

### PR DESCRIPTION
**Pull request by matklad**
* there's no need for `install.sh` when it's just `./scripts/install_zig.sh && ./zig/zig build -Drelease -Dconfig=production`
* there's no need to call `zig build install`, install is the default step
* git hook uses the wrong zig, and in general hooks are very developer-specific

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -179,7 +179,8 @@ First grab the sources and run the setup script:
 ```console
 git clone https://github.com/tigerbeetle/tigerbeetle.git
 cd tigerbeetle
-scripts/install.sh
+./scripts/install_zig.sh
```
Comment (cb22) [inline README.md]: Ah! Let's point the user to the quick start instructions above? They no longer need source to benchmark.

File: .github/ci/benchmark/tigerbeetle.hcl [hunk 2]
```diff
@@ -99,7 +99,7 @@ git clone ${var.git_url}
 cd tigerbeetle
 git checkout ${var.git_ref}
 ./scripts/install_zig.sh
-./zig/zig build install -Drelease -Dconfig=production
+./zig/zig build -Drelease -Dconfig=production
```
Comment (cb22) [inline .github/ci/benchmark/tigerbeetle.hcl]: We can remove this whole folder too, unless you think it's useful to point people to... It's not actively used.

**Comment by cb22** [comment]
(the macOS yaml still references install.sh)

File: .github/ci/benchmark/tigerbeetle.hcl (hunk 2 above)
Comment (matklad) [reply .github/ci/benchmark/tigerbeetle.hcl]: :+1:

File: README.md (hunk 1 above)
Comment (matklad) [reply README.md]: I removed this section altogether. I don't think it's possible to meaningfully use these benchmarks unless you know quite a bit more than we are saying here.

**Review by cb22 — APPROVED** [review]

**Pull request merged**